src/test/java/project/controller/CarControllerTest.java

```diff
@@ -0,0 +1,201 @@
+    void getAllCar_Success() throws Exception {
+        List<CarDto> expectedList = ControllerTestUtil.carDtoListSizeThree();
+
+        MvcResult result = mockMvc.perform(
+                        get("/cars")
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        List<CarDto> actualList = Arrays.stream(
+                objectMapper.readValue(result.getResponse()
+                        .getContentAsByteArray(), CarDto[].class))
+                .toList();
+        assertEquals(expectedList.size(), actualList.size());
+        assertThat(actualList)
+                .usingRecursiveComparison()
+                .withComparatorForType(new BigDecimalComparator(), BigDecimal.class)
+                .ignoringFields("id")
+                .isEqualTo(expectedList);
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"MANAGER"})
+    @Sql(scripts = {DELETE_FROM_CARS, ADD_THREE_CARS},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = DELETE_FROM_CARS,
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Get car by id")
+    void getCarById_Ok() throws Exception {
+        Long id = 3L;
+        CarDto expected = ControllerTestUtil.createCarDto();
+
+        MvcResult result = mockMvc.perform(get("/cars/{id}", id)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        CarDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), CarDto.class);
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"MANAGER"})
+    @Sql(scripts = {DELETE_FROM_CARS,ADD_THREE_CARS},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = DELETE_FROM_CARS,
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void deleteCarById_changeSizeDB_Ok() throws Exception {
+        Long id = 1L;
+        List<CarDto> expected = ControllerTestUtil.carDtoListSizeTwo();
+
+        mockMvc.perform(
+                        delete("/cars/{id}", id)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isNoContent());
+
+        MvcResult result = mockMvc.perform(
+                        get("/cars")
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        List<CarDto> actual = Arrays.stream(
+                objectMapper.readValue(result.getResponse()
+                        .getContentAsString(), CarDto[].class))
+                .toList();
+        assertEquals(expected.size(), actual.size());
+        reflectionEquals(expected,actual);
```

Comment:
always use `assertTrue()` with `reflectionEquals`
```suggestion
        assertTrue(reflectionEquals(expected,actual));
```

Reply:
check other places